### PR TITLE
add more details about the mock implementation

### PR DESCRIPTION
**Pull request by lionel-faber**

### CODE REVIEW

File: Mock-Implementation.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Mock Implementation
+
+## Mock Vault
+
+This is a mock implementation of the [vault](https://github.com/maidsafe/safe_vault) and its various operations:
+
+- Storing data (in a local MockVault file by default, though this can be configured)
+- Authorising operations like reads and mutations
+- Committing mutations and counting them (limit of 1000 by default)
+It can be configured, see [Configuration Parameters](./Mock-vs.-non-mock#configuration-parameters).
```
Comment (mrcnski) [inline Mock-Implementation.md]: Looks like you removed the blank line before this line, which causes it to be included in the previous bullet point when rendering.

File: Mock-Implementation.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+The data stored in the MockVault file can be moved between computers, so an error on an end user's computer can be reproduced locally and investigated by a developer.
+
+The mock vault contains a `Cache` which mocks two authorities, the client manager and the NAE (Network Address Element) manager which manages client accounts and data (Mutable and Immutable) respectively. They hold references to `Account` and `Data` objects mapped against a unique identifier (XorName for Account and DataId for Data). This Cache is held in memory / stored in a file according to the configuration specified.
+
+## Mock Routing
+
+This module has the same interface as [routing](https://github.com/maidsafe/routing) and works with the mock vault instead of vaults. All of the operations are mocked, but made to be as realistic as possible, even incurring some delays via sleep so that they are not nearly-instantaneous.
```
Comment (mrcnski) [inline Mock-Implementation.md]: Looks like you removed the code marks around "sleep" (was `sleep` before).

File: Mock-Implementation.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+## Mock Account
+
+The mock accounts are handled by the mock vault and are what actually count the mutations performed and mutations available. Each mock account has a different counter for these. The mock account also mocks other account properties such as storing the authenticator key and bumping the version when the key is updated.
+
+The client manager authority maps a client's XoR name (i.e. SHA3 hash of the Client's public key) to a mock account object. This object keeps a track of the number of mutations performed and the remaining mutations available along with other account properties such as storing the public key of apps that have been authenticated by a user for an account. The version of the account data is also maintained and validated before every mutation. These properties are tracked separately for each account. Multiple accounts can be created for testing purposes without the need of a valid invitation token. However, to keep the API uniform across mock and non-mock environments the invite token field is still required. On the mock network, any random alphanumeric can be used as an invite token. It will not be validated.
```
Comment (mrcnski) [inline Mock-Implementation.md]: This paragraph repeats some information from the previous paragraph, e.g.:

> This object keeps a track of the number of mutations performed and the remaining mutations available ...

File: Mock-Implementation.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+## Testing
+
+When testing, separate instances of `MockRouting` can be created with custom configuration parameters. For example, to turn off the mutation limit for one test only. This can be done with the `setup_with_config` API which is conditionally compiled only for testing. For an example of its usage see the test `low_balance_check`.
```
Comment (mrcnski) [inline Mock-Implementation.md]: This is an incomplete sentence (has no subject) and does not flow well:

> For example, to turn off the mutation limit for one test only.

Not sure why it was changed, tbh.

**Review by mrcnski — CHANGES_REQUESTED** [review]

File: Mock-Implementation.md (hunk 3 above)
Comment (lionel-faber) [reply Mock-Implementation.md]: Oops! I meant to replace the first para with the second one. Will update :+1:

Comment (mrcnski) [reply Mock-Implementation.md]: One more thing, can this paragraph be broken up as it is a bit long? For example a new paragraph can be started at "Multiple accounts can...", or elsewhere as you see fit.

File: _Sidebar.md [hunk 5]
```diff
@@ -9,6 +9,7 @@
     * [Building for mobile](./Building-for-mobile)
 * [Configuring Client Libs](./Configuring-Client-Libs)
 * [Mock vs. non-mock](./Mock-vs.-non-mock)
+    * [Mock implementation](./Mock-Implementation)
```
Comment (mrcnski) [inline _Sidebar.md]: One final thing: can we change the name of the file to "Mock implementation"? Minor point but it's the convention we've been following (for simplicity and consistency).

**Review by mrcnski — CHANGES_REQUESTED** [review]

File: _Sidebar.md (hunk 5 above)
Comment (mrcnski) [reply _Sidebar.md]: ("implementation" in lowercase)

**Review by mrcnski — APPROVED** [review]

**Pull request merged**